python/lsst/ip/isr/ptcDataset.py

```diff
@@ -0,0 +1,534 @@
+            Other keyword parameters to set in the metadata.
+        """
+        kwargs['PTC_FIT_TYPE'] = self.ptcFitType
+
+        super().updateMetadata(setDate=setDate, **kwargs)
+
+    @classmethod
+    def fromDict(cls, dictionary):
+        """Construct a calibration from a dictionary of properties.
+        Must be implemented by the specific calibration subclasses.
+        Parameters
+        ----------
+        dictionary : `dict`
+            Dictionary of properties.
+        Returns
+        -------
+        calib : `lsst.ip.isr.CalibType`
+            Constructed calibration.
+        Raises
+        ------
+        RuntimeError :
+            Raised if the supplied dictionary is for a different
+            calibration.
+        """
+        calib = cls()
+        if calib._OBSTYPE != dictionary['metadata']['OBSTYPE']:
+            raise RuntimeError(f"Incorrect Photon Transfer Curve dataset  supplied. "
+                               f"Expected {calib._OBSTYPE}, found {dictionary['metadata']['OBSTYPE']}")
+        calib.setMetadata(dictionary['metadata'])
+        calib.ptcFitType = dictionary['ptcFitType']
+        calib.badAmps = np.array(dictionary['badAmps'], 'str').tolist()
+        if calib.ptcFitType in ['FULLCOVARIANCE', ]:
+            # Number of final signal levels
+            nSignalPoints = len(list(dictionary['rawMeans'].values())[0])
+            # The cov matrices are square
+            covMatrixSide = int(np.sqrt(len(
+                                np.array(list(dictionary['aMatrix'].values())[0]).ravel())))
+        for ampName in dictionary['ampNames']:
+            calib.ampNames.append(ampName)
+            calib.inputExpIdPairs.setdefault(
+                ampName, np.array(dictionary['inputExpIdPairs'][ampName]).tolist())
+            calib.expIdMask.setdefault(ampName, np.array(dictionary['expIdMask'][ampName]).tolist())
+            calib.rawExpTimes.setdefault(ampName, np.array(dictionary['rawExpTimes'][ampName]).tolist())
+            calib.rawMeans.setdefault(ampName, np.array(dictionary['rawMeans'][ampName]).tolist())
+            calib.rawVars.setdefault(ampName, np.array(dictionary['rawVars'][ampName]).tolist())
+            calib.gain.setdefault(ampName, np.array(dictionary['gain'][ampName]).tolist())
+            calib.gainErr.setdefault(ampName, np.array(dictionary['gainErr'][ampName]).tolist())
+            calib.noise.setdefault(ampName, np.array(dictionary['noise'][ampName]).tolist())
+            calib.noiseErr.setdefault(ampName, np.array(dictionary['noiseErr'][ampName]).tolist())
+            calib.ptcFitPars.setdefault(ampName, np.array(dictionary['ptcFitPars'][ampName]).tolist())
+            calib.ptcFitParsError.setdefault(
+                ampName, np.array(dictionary['ptcFitParsError'][ampName]).tolist())
+            calib.ptcFitChiSq.setdefault(ampName, np.array(dictionary['ptcFitChiSq'][ampName]).tolist())
+            if calib.ptcFitType in ['FULLCOVARIANCE', ]:
```

Comment:
This looks incorrect.  If the input dictionary contains a field, why would we drop it?

Reply:
If fixed this as we discussed, and the {if} clause is no longer there (wasn't needed).